server/middleware/cacher.js

```diff
@@ -0,0 +1,32 @@
+// This middleware checks for Pokemon Id's and if they are stored in "cache", removes them from the request.
+let cachedIDs = [];
+setInterval(() => {
+   cachedIDs = [];
+}, 60000);
+
+module.exports = function cacher(req, res, next) {
+   if (
+      req.body.task
+         .replace(/\s/g, "")
+         .split(",")
+         .every((elem) => !isNaN(elem))
+   ) {
+      let idsToPass = [];
+      const pokemonIDS = req.body.task.replace(/\s/g, "").split(","); // "1, 2, 3" => [1,2,3]
+      pokemonIDS.forEach((id) => {
+         if (!cachedIDs.includes(id)) {
+            cachedIDs.push(id);
+            idsToPass.push(id);
+         }
+      });
+      if (idsToPass.toString() === "") {
```

Comment:
this check is more accurate (and readble)
```suggestion
      if (idsToPass.length === 0) {
```